packages/million/template.ts

```diff
@@ -194,12 +195,10 @@ export const renderToTemplate = (
     }
 
     canMergeString = false;
-    children += renderToTemplate(child, edits, [...path, k++]);
+    children += renderToTemplate(child, edits, path.concat(k++));
```

Comment:
[benchmark](
https://perf.link/#eyJpZCI6IjU1ZzB3YXFyaDN4IiwidGl0bGUiOiJjb25jYXQgdnMgc2xpY2UucHVzaCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBbLi4uQXJyYXkoNTApLmtleXMoKV0iLCJ0ZXN0cyI6W3sibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJkYXRhLmNvbmNhdCgyKSIsInJ1bnMiOlszNjYwMDAsNTAzMDAwLDk3MTAwMCw5OTQwMDAsNDg1MDAwLDIxMDAwLDIyNjAwMCw1MzIwMDAsNTAwMDAsOTEyMDAwLDI1NzAwMCwyNjUwMDAsODI1MDAwLDU0NTAwMCw2NzYwMDAsMTAxNzAwMCw1MjAwMDAsNjkwMDAwLDg5MzAwMCw1ODAwMDAsNTA5MDAwLDg5NDAwMCwyODQwMDAsMzU2MDAwLDI5ODAwMCwyNjUwMDAsNjU0MDAwLDM0NjAwMCw3NzkwMDAsMTk5MDAwLDMxOTAwMCw2MDUwMDAsMzE1MDAwLDMzNzAwMCwyMTYwMDAsNDk4MDAwLDc1NTAwMCwyMzkwMDAsNjIzMDAwLDI1OTAwMCw1MzQwMDAsNjQyMDAwLDEzNzAwMCwxMDkxMDAwLDIyMTAwMCwxMjUyMDAwLDEwMDAsOTMyMDAwLDExMDkwMDAsMjg5MDAwLDg4MDAwLDYyNDAwMCwyMDAwLDU0NTAwMCw1MTIwMDAsNzgxMDAwLDQ4NTAwMCw1OTAwMCw4MDUwMDAsNjc1MDAwLDI5MDAwLDk2ODAwMCwxODAwMCwzMzMwMDAsMjgxMDAwLDQ3MTAwMCw2NjgwMDAsODMxMDAwLDEyNDYwMDAsNjg3MDAwLDEzNzAwMCwzMzUwMDAsNjA3MDAwLDcxMTAwMCwyMjUwMDAsNDE0MDAwLDgyNzAwMCwxMjE4MDAwLDE5MDAwLDEyMzQwMDAsMTA3NzAwMCwxMTA0MDAwLDYyNDAwMCw1ODcwMDAsMzQxMDAwLDc5MDAwLDQxNzAwMCw4NjEwMDAsNDI1MDAwLDI0MDAwMCwxMTU0MDAwLDEyMzUwMDAsMTY1MDAwLDEyNjAwMDAsODI1MDAwLDEzOTAwMDAsMTUzMDAwLDQ2ODAwMCwxMTMzMDAwLDEwNjQwMDBdLCJvcHMiOjU2NzIzMH0seyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6IlsuLi5kYXRhLCAyXSIsInJ1bnMiOlszMzIwMDAsMjQ1MDAwLDEwMDAsMTk1MDAwLDEwMzIwMDAsMjAwMDAsMTMwMDAwLDY5ODAwMCwyMDAwMCwzODkwMDAsOTA1MDAwLDEwOTkwMDAsMjM5MDAwLDUyMTAwMCw4ODUwMDAsNjkwMDAsMjMwMDAsNzcxMDAwLDEwNTkwMDAsNTg0MDAwLDg2NTAwMCwxMDMyMDAwLDI3NzAwMCw0MTQwMDAsMzA0MDAwLDQzNzAwMCw0NTEwMDAsMzcxMDAwLDgyMDAwLDIwMDAwLDE0ODAwMCwyNzcwMDAsMjcxMDAwLDYxODAwMCwzOTgwMDAsNjMxMDAwLDEzNjAwMCw4MTIwMDAsNTU1MDAwLDI0MDAwLDEyNjgwMDAsNTk4MDAwLDc4MTAwMCwxMTUxMDAwLDIxNzAwMCw4NDUwMDAsMTAwMCw4NDUwMDAsNTI0MDAwLDI4OTAwMCwxMTE2MDAwLDU3ODAwMCw4NTgwMDAsNjM0MDAwLDU3ODAwMCwxMTEwMDAsMTAzNTAwMCw5ODUwMDAsMjAwMDAsNTUyMDAwLDgwNDAwMCw4ODgwMDAsMTY0MDAwLDEyODYwMDAsNTc4MDAwLDQwOTAwMCwxMjYxMDAwLDY4NjAwMCw2NjAwMCwxNjQwMDAsODkxMDAwLDc3NTAwMCwxMjM0MDAwLDI3NzAwMCwyNTEwMDAsNzUwMDAwLDg0NTAwMCwxMzg5MDAwLDQ1NzAwMCw2ODUwMDAsOTMyMDAwLDE3MDAwLDEwMDQwMDAsMTEyNjAwMCwzMDAwMCwxMDAwLDUwNDAwMCw2NjUwMDAsNzA1MDAwLDc4MjAwMCwyMjAwMDAsMTA4NTAwMCwxMjEzMDAwLDk4NTAwMCw1MzUwMDAsMTM1MjAwMCw5OTIwMDAsMTIzMDAwLDc5ODAwMCwxMTg1MDAwXSwib3BzIjo1ODQ0MDB9LHsibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJkYXRhLnNsaWNlKCkucHVzaCgyKSIsInJ1bnMiOlszNTIwMDAsMjQyMDAwLDEzNjYwMDAsMTA2NzAwMCwzNTcwMDAsMTIxMDAwLDgzMjAwMCw2MTgwMDAsNTQwMDAsMTcwMDAsNTAwMDAsMTA0NTAwMCwxOTEwMDAsNTE4MDAwLDg3ODAwMCwxMTYwMDAsMTAwMCwxMzEwMDAsMTAwNTAwMCw1MTAwMDAsODY1MDAwLDc5MTAwMCwyODAwMDAsMzkyMDAwLDMwNDAwMCwyNDkwMDAsNDkwMDAwLDM5NzAwMCw0NzMwMDAsMzkwMDAwLDI1MTAwMCwzNDQwMDAsMjkxMDAwLDUyNTAwMCwzMTAwMDAsMTU0MDAwMCwzMDQwMDAsNzgwMDAwLDc0MDAwMCwxMTM1MDAwLDMzNzAwMCwzOTAwMDAsMTU0NjAwMCwxMDAyMDAwLDIzOTAwMCw0NzEwMDAsMTUwNjAwMCwxMTE5MDAwLDQ1ODAwMCwxMDAwMCwxMDAwMCwxNTQ2MDAwLDE3MDAwLDIyNjAwMCwyNTgwMDAsNjA3MDAwLDEyMTIwMDAsNDI3MDAwLDU3MDAwLDQ5NjAwMCwyMTkwMDAsMTEwNjAwMCwxMTgwMDAwLDM0NDAwMCwxMTkyMDAwLDEyMDUwMDAsMzU3MDAwLDc5MTAwMCw0NzEwMDAsNDI3MDAwLDI5NDAwMCwyNzEwMDAsOTI1MDAwLDQ3MTAwMCwyMjEwMDAsMTAxMzAwMCwxMDUwMDAsNDAyMDAwLDExNjcwMDAsNTUxMDAwLDEwNDUwMDAsNDk3MDAwLDI0MDAwLDMxMDAwMCw5OTgwMDAsOTcwMDAsMzQ0MDAwLDY3MTAwMCw3MjQwMDAsMTg3MDAwLDMxODAwMCwyODYwMDAsMjUzMDAwLDM3NzAwMCw2OTQwMDAsMTcwMDAsMTcwMDAsMTkzMDAwLDEwMTkwMDAsMTIzODAwMF0sIm9wcyI6NTQyNzcwfV0sInVwZGF0ZWQiOiIyMDIzLTA0LTEyVDAyOjI1OjQ3LjEzNloifQ%3D%3D)

**Chrome**

<img width="1457" alt="image" src="https://user-images.githubusercontent.com/40715044/231332216-e2de8bfc-1248-4c0d-aa35-69feddd90928.png">

**Firefox**

<img width="1348" alt="image" src="https://user-images.githubusercontent.com/40715044/231332143-577d58b5-8eb8-4e4c-b0f4-77af93d2e681.png">

**Safari**

<img width="1315" alt="image" src="https://user-images.githubusercontent.com/40715044/231332398-4cf66721-13f1-4d9d-a8da-ee0a70949ca2.png">

Reply:
Nice check, I thought concat would be more efficient because it didn't need to iterate but I guess not. Switching to the slice implementation

Reply:
> Nice check, I thought concat would be more efficient because it didn't need to iterate but I guess not. Switching to the slice implementation

FYI on Chrome `slice().push()` is the slowest.

Reply:
Yes, maybe we can switch off the option at the inflection point to the first implementation in [this benchmark](https://perf.link/#eyJpZCI6ImNnb3h4YjF3ZHV5IiwidGl0bGUiOiJjb25jYXQgdnMgc2xpY2UucHVzaCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBbLi4uQXJyYXkoMTAwMCkua2V5cygpXVxuY29uc3QgY29uY2F0JCA9IEFycmF5LnByb3RvdHlwZS5jb25jYXQ7XG5jb25zdCBzbGljZSQgPSBBcnJheS5wcm90b3R5cGUuc2xpY2U7XG5jb25zdCBwdXNoJCA9IEFycmF5LnByb3RvdHlwZS5wdXNoOyIsInRlc3RzIjpbeyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6ImNvbmNhdCQuYXBwbHkoW10sIFtkYXRhLCBbMl1dKTsiLCJydW5zIjpbMTI2MDAwLDIyMDAwMCwyMjAwMDAsMjIwMDAwLDIyMDAwMCwzNjEwMDAsMjIwMDAwLDM0OTAwMCwxMjUwMDAsNDQ3MDAwLDIyMDAwMCwyMDQwMDAsMTAwMCw0NTQwMDAsNjYwMDAsMzc3MDAwLDE5MjAwMCwyMjAwMDAsMTcwMDAsNDQ3MDAwLDQ0NzAwMCwzOTcwMDAsMTYwMDAwLDUwMDAwLDQ0NzAwMCwyMjAwMDAsMjIwMDAwLDQwNjAwMCwyMDAwMDAsODYwMDAsMjIwMDAwLDQxNTAwMCw5MDAwLDIyMDAwMCwzMzIwMDAsMTkyMDAwLDM0MzAwMCwxNzAwMCwxMjYwMDAsMjIwMDAwLDIyMDAwMCwxNjAwMCwyMjAwMDAsMjA1MDAwLDQxMTAwMCwxOTEwMDAsNDQ3MDAwLDE3NjAwMCwyNzEwMDAsMjY1MDAwLDE2MzAwMCwyMjAwMDAsMjU2MDAwLDE3ODAwMCwzNTMwMDAsMjIwMDAwLDM4ODAwMCwyMjAwMDAsMTQyMDAwLDE2ODAwMCwzOTAwMCwxMTQwMDAsMjk5MDAwLDIwNDAwMCwyMjAwMDAsMjIwMDAwLDIyMDAwMCwyMjAwMDAsMjIwMDAwLDI4MTAwMCwxNTUwMDAsNTAwMDAsMjUyMDAwLDcyMDAwLDc0MDAwLDE0NzAwMCwxODQwMDAsMjI5MDAwLDIyMDAwMCwxMDAwLDIyMDAwMCwyMjAwMDAsMjIxMDAwLDU1MDAwLDc4MDAwLDMzNDAwMCwyNDAwMCwxMTQwMDAsMzE3MDAwLDIyMDAwMCw5MjAwMCwzMzMwMDAsMjIwMDAwLDIyMDAwMCwyMjAwMDAsMjIwMDAwLDI3NzAwMCwyMjAwMDAsMTYwMDAsNDQzMDAwXSwib3BzIjoyMTY0ODB9LHsibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiJkYXRhLmNvbmNhdCgyKSIsInJ1bnMiOlszMjAwMCw1NzAwMCw5MDAwMCwyMjAwMCwyOTAwMCwxMDAwMCw0NzAwMCwxMDAwMDAsMTkwMDAsMTQ4MDAwLDMxMDAwLDg2MDAwLDEzNDAwMCwxODEwMDAsMTUwMDAsMTM2MDAwLDMxMDAwLDQ5MDAwLDgwMDAwLDE1MTAwMCw2NTAwMCwxMjcwMDAsMTE3MDAwLDMxMDAwLDE2OTAwMCw4MzAwMCw5NDAwMCwyNjAwMCw0MzAwMCw0ODAwMCw5MDAwLDU2MDAwLDE4MDAwLDk4MDAwLDk4MDAwLDg4MDAwLDExNjAwMCwxNDQwMDAsNzEwMDAsMjUwMDAsMzkwMDAsMTAwMCw5NDAwMCw0NDAwMCwxMDgwMDAsMTAxMDAwLDExOTAwMCwzNTAwMCwxMjQwMDAsMTMyMDAwLDIzMDAwLDIzMDAwLDU4MDAwLDc4MDAwLDEzMDAwMCw1MzAwMCw1MDAwMCw0MjAwMCwxNDAwMCwxMTAwMCw3MDAwLDEwMDAwLDQ4MDAwLDIwMDAwLDMyMDAwLDY1MDAwLDIzMDAwLDc1MDAwLDUwMDAwLDEzMDAwLDUyMDAwLDUzMDAwLDQ5MDAwLDk1MDAwLDQzMDAwLDcwMDAsMTgwMDAsOTkwMDAsMjYwMDAsMTAzMDAwLDQ0MDAwLDI3MDAwLDMxMDAwLDIwMDAsOTUwMDAsNzEwMDAsNDgwMDAsODMwMDAsMTIwMDAsNjcwMDAsNDAwMDAsODMwMDAsOTEwMDAsMjAwMDAsODIwMDAsNjAwMDAsODEwMDAsNzIwMDAsMTEwMDAsOTIwMDBdLCJvcHMiOjYyNTMwfSx7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoiWy4uLmRhdGEsIDJdIiwicnVucyI6WzM1MDAwLDYwMDAsOTAwMDAsNzAwMDAsMzgwMDAsMTYxMDAwLDgwMDAwLDI1MDAwLDE3NTAwMCw5MDAwMCwxNDAwMCwxODUwMDAsMTgzMDAwLDE4MzAwMCwxNjMwMDAsOTAwMDAsNTAwMDAsMzgwMDAsOTAwMDAsMTg2MDAwLDkwMDAwLDE2ODAwMCwzMzgwMDAsMTIwMDAsMTgyMDAwLDkwMDAwLDg5MDAwLDkwMDAwLDkwMDAwLDE0NzAwMCwxNDYwMDAsMTE1MDAwLDE1MjAwMCwxMDkwMDAsMTE2MDAwLDkwMDAwLDE4NTAwMCw0MTAwMCw0MTAwMCw2NzAwMCw5MDAwMCwxMTEwMDAsOTAwMDAsMTAwMDAwLDkwMDAwLDkwMDAwLDEwNTAwMCw5MDAwMCwxMTYwMDAsMTI1MDAwLDc1MDAwLDE4MjAwMCwxNTIwMDAsMTc1MDAwLDM4MDAwLDkwMDAwLDkwMDAwLDE0OTAwMCwxMzIwMDAsOTAwMDAsMTAyMDAwLDEwNDAwMCwxNTAwMCw1OTAwMCw2MjAwMCwxMTgwMDAsNzUwMDAsOTAwMDAsMzQwMDAsMTAyMDAwLDkwMDAwLDExODAwMCw2MzAwMCwxMDkwMDAsMjEwMDAsMTI3MDAwLDE4MDAwLDE0NjAwMCw5MDAwMCw5MDAwMCwxMjQwMDAsOTAwMDAsMTgyMDAwLDEwMzAwMCwzMjAwMCw1ODAwMCw5MDAwMCwxMzQwMDAsOTUwMDAsOTAwMDAsMTc1MDAwLDEzMzAwMCw5MDAwMCw5MDAwMCw5MDAwMCw5MDAwMCw0NTAwMCw5MDAwMCw5MDAwMCw1NDAwMF0sIm9wcyI6MTAwNjgwfSx7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoicHVzaCQuY2FsbChzbGljZSQuY2FsbChkYXRhKSwgMik7IiwicnVucyI6WzExNDAwMCwyOTcwMDAsOTAwMDAsNDAwMDAsMjM4MDAwLDE3NDAwMCwyMzAwMDAsOTAwMDAsODAwMDAsNDEwMDAsMTM5MDAwLDE1MDAwLDkwMDAwLDQwMDAsOTAwMDAsMzAwMCwxMjYwMDAsMzIwMDAsODAwMDAsMTgzMDAwLDkwMDAwLDkxMDAwLDEwMDAsNTAwMCwyMTUwMDAsNDAwMCw5MDAwMCwxNDAwMDAsMTE3MDAwLDI2MDAwLDkwMDAwLDkwMDAwLDEwMDAsMjkzMDAwLDkwMDAwLDkwMDAwLDEyMzAwMCw2MjAwMCw2NzAwMCwxODAwMCw2NjAwMCw2MzAwMCw5MDAwMCw5MDAwMCw0MjAwMCwxMzcwMDAsMTgzMDAwLDE3ODAwMCw5MDAwMCw5MDAwMCwyNjkwMDAsMzcwMDAsNTgwMDAsMzYwMDAsODAwMDAsMTMwMDAwLDU4MDAwLDM1MDAwLDc0MDAwLDkwMDAwLDkwMDAwLDMwMDAwLDMwMDAwLDIxMDAwLDU2MDAwLDkwMDAwLDI5MDAwLDUyMDAwLDkwMDAwLDE3MDAwLDEzNDAwMCw0NjAwMCwxMTYwMDAsODUwMDAsNDAwMDAsMTEzMDAwLDgwMDAwLDEwMzAwMCwyMTYwMDAsMTI5MDAwLDMyMDAwLDQ4MDAwLDE4MzAwMCw5MDAwMCwxNjEwMDAsOTYwMDAsOTAwMDAsOTAwMDAsNDEwMDAsNjQwMDAsOTAwMDAsMTE4MDAwLDU2MDAwLDE4MzAwMCwxOTAwMCwxODMwMDAsNDEwMDAsODUwMDAsMTE0MDAwLDI4MzAwMF0sIm9wcyI6OTM4OTB9XSwidXBkYXRlZCI6IjIwMjMtMDQtMTJUMDI6NDM6NDQuODg0WiJ9):

Reply:
@aidenybai 

An array size of 10 has a very different result:

<img width="1824" alt="image" src="https://user-images.githubusercontent.com/40715044/231340029-4b015505-81bc-4517-98a4-414ea57c1d5b.png">

Might need to take a consideration into that.

Reply:
Switcher code should be implemented

Reply:
> Switcher code should be implemented

IMHO, it is uncommon for the `path` to have a size of more than 10.

Reply:
good point, then we can just remove the switcher. imo here this is a pretty small microoptimization and seems like .slice() works generally well on all browsers, so switching to that should be okay